yoke/assets/joypad/base.js

```diff
@@ -296,61 +443,73 @@ function Joypad() {
             this._controls.push(new Dummy('dum', updateStateCallback));
         }
     }
-    // end of section to be deleted
+    // End of section to be deleted.
 
-    this._debugLabel = new Control('debug', 'dbg');
-    joypad.appendChild(this._debugLabel.element);
 }
 Joypad.prototype.updateState = function() {
     var state = this._controls.map(function(control) { return control.state(); }).join(',');
 
     Yoke.update_vals(state); // only works in yoke webview
 
     this._debugLabel.element.innerHTML = state;
-    if (!DEBUG_NO_CONSOLE_SPAM) {
-        console.log(state);
-    }
+    if (!DEBUG_NO_CONSOLE_SPAM) { console.log(state); }
 };
 
 //
 // BASE CODE
 //
+// These variables are automatically updated by the code
 var joypad = null;
 var buttons = 0; // number of buttons total
 var axes = 0; // number of analog axes total
+var motionState = [0, 0, 0, 0, 0, 0];
+var motionSensor = null;
+var deviceMotionEL = null;
+var deviceOrientationEL = null;
 
-window.addEventListener('resize', function() {
-    if (joypad != null) {
-        joypad._controls.forEach(function(control) {
-            control.onAttached();
-        });
-    }
-});
+// If the user's browser needs permission to vibrate
+// it's more convenient to ask for it first before entering fullscreen.
+window.navigator.vibrate(50);
```

Comment:
I think `navigator.vibrate()` cannot run on page load, only as a response to some event (just like fullscreen request). I haven't seen a mobile browser that asks user to permit vibration, though.

Reply:
Firefox for Android asks when the website first tries to run `navigator.vibrate()` (in this test, as soon as I touch a joystick or button):
![Allow me to vibrate](https://user-images.githubusercontent.com/47781797/55839236-abb9cf00-5b16-11e9-911d-da5b59a23ea5.png)

And from what I checked in Firefox, `navigator.vibrate` has no such restriction. Fullscreen is restricted to prevent a website from hijacking the screen, but the vibration API isn't nearly as dangerous or annoying.

Reply:
Only Github support can remove images. I can see the old image in my email inbox.

Firefox asks whether it should allow vibration only AFTER something was tapped, so it's a response to some `onTouchStart`. It doesn't ask on page load, because vibrating on page load is not allowed.

Chrome doesn't ask for vibration permission, yet it also does not vibrate on page load.

When vibration API came out, many crappy ad sites abused it (e.g. "Your phone has been hacked, Google confirmed, download this APK to fix your system" with phone vibrating all the time)

Reply:
Okay, nvm, I'm wrong, I was testing the wrong branch

Reply:
As you can see, everyone makes mistakes